### PR TITLE
adding project in

### PR DESCRIPTION
**Pull request by Baumelbi**
I expanded on my initial investigations into pandas library and created a heat map generator which can find the users location based on their public IP or a manually entered location within Seattle

### CODE REVIEW

**Comment by PythonCHB** [comment]
Is it ready for review?

-CHB

**Comment by Baumelbi** [comment]
Not quite yet I should have waited for the pull request

**Comment by PythonCHB** [comment]
no problem.

**Comment by Baumelbi** [comment]
I wouldn't mind a review of the code at this time. i've made quite a few changes since my last pull request.

**Comment by Baumelbi** [comment]
My project is ready to be reviewed. Brad Baumel

File: students/baumel/project/crimeHeatMap.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
+"""This program will call to the seattle 911 api and recieve information realting to 
+the given latitude and longitude as well as the radius of the circle. The returned
+information will be ranked by most common instance to least common
+
+This app will then grab the users current location and use the gmplot library to generate
+a heat map of the crime data the user will also have the option to enter in a location
+within seattle"""
+
```
Comment (PythonCHB) [inline students/baumel/project/crimeHeatMap.py]: cool stuff!

File: students/baumel/project/crimeHeatMap.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	
+	return location
+
+def radiusAndLimit():
+	goodLimit = False
+	goodRadius = False
+
+	while (not goodLimit):
+		limit = input("How many results would you like? (1000 - 50,000) ")
+
```
Comment (PythonCHB) [inline students/baumel/project/crimeHeatMap.py]: a style issue -- this is a lot of whitespace!

File: students/baumel/project/crimeHeatMap.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	goodRadius = False
+
+	while (not goodLimit):
+		limit = input("How many results would you like? (1000 - 50,000) ")
+		
+		try:
+			int(limit)
+		except ValueError:
+			print ("Invalid value entered!")
+		else:
```
Comment (PythonCHB) [inline students/baumel/project/crimeHeatMap.py]: nice use of else in a try block!

File: students/baumel/project/crimeHeatMap.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+			limit = int(limit)
+			if limit >= 1000 and limit <= 50000:
```
Comment (PythonCHB) [inline students/baumel/project/crimeHeatMap.py]: really minor, but you could have put this line:

limit = int(limit) above in the try -- and then you wouldn't need to call it again here.

File: students/baumel/project/crimeHeatMap.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+	else:
+		location = manualLocation(geolocator)
+
+	while location is None or "seattle" not in location.address.lower():
+			location = manualLocation(geolocator)
+	
+	latitude = geolocator.geocode(location).latitude
+	longitude = geolocator.geocode(location).longitude
+	address = geolocator.geocode(location).address
+	radius, limit = radiusAndLimit()
```
Comment (PythonCHB) [inline students/baumel/project/crimeHeatMap.py]: better to call geocode once, and then extract the data:

place = geolocator.geocode(location)

latitude = place.latitude
longitude = place.longitude
...

or do that bit when building the dict.

Or even forget the dict, and just pass the object that geocode returns...

File: students/baumel/project/crimeHeatMap.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+	# taking the parsed JSon and alligning it with the map
+	gmap = gmplot.GoogleMapPlotter(locationInfo['latitude'],locationInfo['longitude'], 13)
+	gmap.heatmap(raw_data["latitude"], raw_data["longitude"], radius = 30)
+	gmap.draw("mymap.html")
+
+
+def output(raw_data, locationInfo):
+	# opens up the html file in your default browser
+	url = "mymap.html"
+	webbrowser.open('file://' + os.path.realpath(url))
```
Comment (PythonCHB) [inline students/baumel/project/crimeHeatMap.py]: nice fiind -- prety handy, eh?

**Pull request merged**